### PR TITLE
Separate LabelScanStore tests from actual label scan store implementation

### PR DESCRIPTION
**Pull request by burqen**
This is preparation for letting native lss use same tests as current implementation.

### CODE REVIEW

File: community/kernel/src/test/java/org/neo4j/kernel/api/impl/labelscan/LabelScanStoreTest.java [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+    {
+        life.shutdown();
+    }
+
+    protected abstract LabelScanStore createLabelScanStore( FileSystemAbstraction fileSystemAbstraction,
+            File rootFolder, List<NodeLabelUpdate> existingData, boolean usePersistentStore, boolean readOnly,
+            LabelScanStore.Monitor monitor );
+
+    @Test
+    public void failToRetrieveWriterOnReadOnlyScanStore() throws IOException
```
Comment (MishaDemianenko) [inline community/kernel/src/test/java/org/neo4j/kernel/api/impl/labelscan/LabelScanStoreTest.java:115]: `failToRetrieveWriterOnReadOnlyScanStore` and `failToGetWriterOnReadOnlyScanStore` are the same

File: community/kernel/src/test/java/org/neo4j/kernel/api/impl/labelscan/LabelScanStoreTest.java [hunk 2]
```diff
@@ -0,0 +1,640 @@
...
+        expectedException.expect( UnsupportedOperationException.class );
+        store.newWriter();
+    }
+
+    @Test
+    public void failToStartIfLabelScanStoreIndexDoesNotExistInReadOnlyMode()
+    {
+        expectedException.expectCause( Matchers.instanceOf( UnsupportedOperationException.class ) );
+        start( false, true );
+//        assertTrue( monitor.noIndexCalled );
```
Comment (MishaDemianenko) [inline community/kernel/src/test/java/org/neo4j/kernel/api/impl/labelscan/LabelScanStoreTest.java]: leftovers

**Review by MishaDemianenko — COMMENTED** [review]
2 small comments, duplicates are actually even where there before.
Otherwise, ready to merge on green

**Comment by burqen** [comment]
@MishaDemianenko Nice! Addressed 👍

**Pull request merged**